lasy/profiles/speckle_profile.py

```diff
@@ -0,0 +1,371 @@
+
+    The speckles are created using with random (RPP) or continuous phase plates (CPP)
+    then either the Smoothing by spectral dispersion (SSD)
+    or Induced spatial incoherence methods.
+    More precisely, the electric field corresponds to:
+
+    .. math::
+
+        E_u(\boldsymbol{x}_\perp,t) &= Re\left[ E_0\,
+        \sum_{j=1}^{N_{bx}\times N_{by}} A_j
+        {\rm sinc}\left(\frac{\pi D_xx}{\lambda_0 f}\right)
+        {\rm sinc}\left(\frac{\pi D_yy}{\lambda_0 f}\right)\\\\
+        \exp\left(i\boldsymbol{k}_{\perp,j}\cdot\boldsymbol{x}_\perp
+        + i\phi_{{\rm RPP/CPP},j}+i\psi_{{\rm SSD/ISI},j}\right) \times p_u \right]
+
+    where :math:`u` is either :math:`x` or :math:`y`, :math:`p_u` is
+    the polarization vector, :math:`Re` represent the real part, and
+    :math:`\boldsymbol{x}_\perp=(x,y)` is the transverse coordinate (orthogonal
+    to the propagation direction).
+    Several quantities are computed internally to the code depending on the
+    method of smoothing chosen, including the beamlet amplitude :math:`A_j`,
+    the beamlet wavenumber at focus :math:`k_{\perp,j}`,
+    the phase contribution :math:`\phi_{{\rm RPP/CPP},j}` from the phase plate,
+    and the phase contribution :math:`\psi_{{\rm SSD/ISI},j}` from the smoothing.
+    The other parameters in this formula are defined below.
+
+    Notes
+    -----
+    This assumes a rectangular laser and so a rectangular grid of beamlets.
+
+    Parameters
+    ----------
+    wavelength : float (in meter)
+        The main laser wavelength :math:`\lambda_0` of the laser, which
+        defines :math:`\omega_0` in the above formula, according to
+        :math:`\omega_0 = 2\pi c/\lambda_0`.
+
+    pol : list of 2 complex numbers (dimensionless)
+        Polarization vector. It corresponds to :math:`p_u` in the above
+        formula ; :math:`p_x` is the first element of the list and
+        :math:`p_y` is the second element of the list. Using complex
+        numbers enables elliptical polarizations.
+
+    w0 : float (in meter)
+        The waist of the laser pulse, i.e. :math:`w_0` in the above formula.
+
+    tau : float (in second)
+        The duration of the laser pulse, i.e. :math:`\tau` in the above
+        formula. Note that :math:`\tau = \tau_{FWHM}/\sqrt{2\log(2)}`,
+        where :math:`\tau_{FWHM}` is the Full-Width-Half-Maximum duration
+        of the intensity distribution of the pulse.
+
+    t_peak : float (in second)
+        The time at which the laser envelope reaches its maximum amplitude,
+        i.e. :math:`t_{peak}` in the above formula.
+
+    focal_length : float (in meter)
+        Focal length of lens :math:`f` just after the RPP/CPP.
+
+    beam_aperture : list of 2 floats (in meters)
+        Beam width :math:`D_x,D_y` at the lens / size of the illuminated region of the RPP/CPP.
+
+    n_beamlets : list of integers
+        Number of RPP/CPP elements :math:`N_{bx},N_{by}` in each direction.
+
+    lsType : string
```

Comment:
Could we make the name of this argument more explicit?
```suggestion
    longitudinal_smoothing_type : string
```
(I assume that this is what `ls` means here.)

Reply:
If we do adopt this name, it should be changed throughout the code and throughout this docstring.

Reply:
Maybe a better name would be `temporal_smoothing_type`?